floris/wind_data.py

```diff
@@ -1121,6 +1126,14 @@ def __init__(
         if not isinstance(turbulence_intensities, np.ndarray):
             raise TypeError("turbulence_intensities must be a NumPy array")
 
+        # Check that wind_directions, wind_speeds, and turbulence_intensities do not contain NaNs
+        if np.isnan(wind_directions).any():
+            raise ValueError("wind_directions must not contain NaNs")
+        if np.isnan(wind_speeds).any():
+            raise ValueError("wind_speeds must not contain NaNs")
+        if np.isnan(turbulence_intensities).any():
+            raise ValueError("turbulence_intensities must not contain NaNs")
```

Comment:
I propose we place these closer to the checks for monotonicity (or possibly simply include them in the same error raised)